File: java-advent-of-code/src/test/java/advent/twenty_seventeen/Day08IHeardYouLikeRegistersTest.java
package advent.twenty_seventeen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.*;

public class Day08IHeardYouLikeRegistersTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day08IHeardYouLikeRegisters("/twenty_seventeen/Day08-IHeardYouLikeRegisters-input.txt");

        assertEquals(5221, dp.getPart1Answer().intValue());
        assertEquals(7491, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day08IHeardYouLikeRegisters("/twenty_seventeen/Day08-IHeardYouLikeRegisters-example#1.txt");

        assertEquals(1, dp.getPart1Answer().intValue());
        assertEquals(10, dp.getPart2Answer().intValue());
    }

}

File: scala-advent-of-code/src/test/scala/advent/twenty_seventeen/Day08IHeardYouLikeRegistersTest.scala
package advent.twenty_seventeen

import org.scalatest.{FlatSpec, Matchers}

class Day08IHeardYouLikeRegistersTest  extends FlatSpec with Matchers {
  "2017 Day 08 - Input" should "calculate answers" in {
    val dp = new Day08IHeardYouLikeRegisters("/twenty_seventeen/Day08-IHeardYouLikeRegisters-input.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }

  "2015 Day 9 - Example #1" should "calculate answers" in {
    val dp = new Day08IHeardYouLikeRegisters("/twenty_seventeen/Day08-IHeardYouLikeRegisters-example#1.txt")
    dp.part1Answer should be(0)
    dp.part2Answer should be(0)
  }
}
